python/python/lance/cleanup.py

```diff
@@ -0,0 +1,31 @@
+#  Copyright (c) 2023. Lance Developers
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import List, Tuple
+
+from .lance import _cleanup_partial_writes
+
+
+def cleanup_partial_writes(objects: List[Tuple[str, str]]):
```

Comment:
this input `objects` is tight to one type of `FileSystemFragmentWriteProgress` implementation.

Reply:
What do you mean? It's tightly coupled with the object-store implementation (using multi-part uploads), not the progress. For the same object store impl, what other API could we have?

This tight coupling was why I was saying earlier we should just use vacuum instead of implementing progress and cleanup.

Reply:
For example, if say there is an implementation that stores `progress` on Redis or Dynamodb, or some JobHistoryServer in some frameworks, we would want to clean more than the half-write files, as well as the marker of the progress.

Or just every such implementation needs to make a new `cleanup_partial_write()`.  Can we make a `FragmentWriteProgress::abort()` API or so, so that `cleanup_partital_write` is orthogonal to the implementation?

Reply:
Yeah that makes sense to put the cleanup in the progress tracker. 👍 

I don’t think I fully realized the scope of what you were thinking of when you made the progress tracker. That makes sense. It sounds like we’ll be doing a lot more work to integrate Lance with distributed frameworks.